src/api/helpers/ApiHelpers.ts

```diff
@@ -0,0 +1,69 @@
+import { RequestConfigType } from "./ApiRequestTypes";
+import { DEFAULT_ERROR_MESSAGE, MAIN_API_URLS, METHODS } from "./ApiConstant";
+
+const generateEndpointUrl = (url:string):string => MAIN_API_URLS.BASE+MAIN_API_URLS.VERSION+url;
+
+const generateHeaders = (token: string | undefined) => {
+  const headers: { [key: string]: string } = {
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  };
+  if (token) {
+    headers.Authorization = `Bearer ${token}`;
+  }
+  return headers;
+};
+
+const generateRequestConfig = ({ method, token, params }: RequestConfigType) => {
+  switch (method) {
+    case 'GET':
+    case 'DELETE':
+      return {
+        method,
+        headers: generateHeaders(token),
+      };
+    case 'PUT': {
+      if (params) {
+        return {
+          method,
+          headers: generateHeaders(token),
+          body: JSON.stringify(params),
+        };
+      }
+      return {
+        method,
+        headers: generateHeaders(token),
+      };
+    }
+    case 'POST':
+      return {
+        method,
+        headers: generateHeaders(token),
+        body: JSON.stringify(params),
+      };
+    default:
+      return {
+        method,
+        headers: generateHeaders(token),
+      };
+  }
+};
+
+const fetchMainAPI = async (url: string, requestConfig: RequestConfigType) => {
+  const response = await fetch(generateEndpointUrl(url), generateRequestConfig(requestConfig));
+  const json = await response.json();
+  if (!response.ok) throw new Error((await json.message) || DEFAULT_ERROR_MESSAGE);
```

Comment:
Better way if we check or show status code and status text with our message. I don't know how works backend, so this is just proposal:
```
if (!response.ok) throw new Error((json.message || `${response.status} ${response.statusText} at ${url}`) || DEFAULT_ERROR_MESSAGE);
```

And better way if we have different messages for different errors. Because sometimes its not just Server request error and we (and user) should know about it.

Reply:
Block with try catch in "fetchMainAPI" function. I suppose get  message text of error from back.